### PR TITLE
added docstrings to generic models for API docs

### PR DESCRIPTION
**Pull request by agladstein**
closes https://github.com/popgensims/stdpopsim/issues/180
closes https://github.com/popgensims/stdpopsim/issues/181

I added what made sense to me. Let me know if there is anything else that you think is necessary for the API docs on those models. 
I'm also not sure if those parameters should actually all be floats, or maybe some are supposed to be ints?

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Very nice! Just needs a quick lint-fix on line length.

**Comment by agladstein** [comment]
> Very nice! Just needs a quick lint-fix on line length.

I left that one line over because it's the python example, and thought it looked weird if I made it a multi-liner.
Could change it to:
```
    :Example:
        >>> # One change
        >>> model1 = stdpopsim.PiecewiseConstantSize(N0, (t1, N1)) 
        >>> # Two changes
        >>> model2 = stdpopsim.PiecewiseConstantSize(N0, (t1, N1), (t2, N2)) 
```

**Comment by jeromekelleher** [comment]
Is ``:Example:`` sphinx markup? I haven't seen this used before. We could put these in as a code block, which would get rid of the ``>>>`` at the start.

If you do want the docstring to be too long, you need a ``#noqa`` at the end of it like

```python
def something():
    ```
    Docstring telling me something about something.
    ```   # noqa
    # rest of something function
```

**Comment by agladstein** [comment]
> Is `:Example:` sphinx markup? 
Not sure, I sure. It's just what I found that worked.

**Comment by jeromekelleher** [comment]
Perfect; can you rebase and [squash](https://stdpopsim.readthedocs.io/en/latest/development.html#rebasing) this down to one commit please?

**Comment by agladstein** [comment]
Thank you for the rebasing docs! worked perfectly 😄

File: stdpopsim/models.py [hunk 1]
```diff
@@ -243,6 +243,24 @@ def simulate(self, contig, samples, seed=None):
...
 class PiecewiseConstantSize(Model):
+    """
+    Class representing a generic simulation model that can be run to output a
+    tree sequence. This is a piecewise constant size model, which allows for
+    instantaneous population size change over multiple epochs in a single population.
+
+    :ivar N0: The initial effective population size
+    :vartype N0: float
+    :ivar args: Each subsequent argument is a tuple (t, N) which gives the
+        time at which the size change takes place and the time.
```
Comment (jeromekelleher) [inline stdpopsim/models.py]: Just spotted a typo here: should be "takes place and the population size".

**Comment by jeromekelleher** [comment]
Minor typo spotted above.

In case you don't know, ``git commit --amend`` is handy for this type of thing. Make the change, do git commit -a --amend and force push. A bit quicker than creating another commit and then squashing that out.

**Comment by agladstein** [comment]
Good find. Thanks for the suggestion.

**Pull request merged**